serverlessworkflow/modules/ROOT/pages/service-orchestration/orchestration-of-asyncapi-based-services.adoc

```diff
@@ -0,0 +1,166 @@
+
+.Example of an AsynyAPI subscribing channel definition
+[source,yaml]
+----
+channels:
+  wait:
+    description: A message channel
+    subscribe:
+      operationId: consumeWait
+      summary: Get messages
+      message:
+        $ref: '#/components/messages/message'
+----
+
+You can define a function in your workflow, of type asyncapi, that points to `consumerWait` operationId.
+
+.Example of an AsynyAPI subscribing function definition
+[source,json]
+----
+{
+   "functions":[
+      {
+          "name": "consumeEvent",
+          "type": "asyncapi",
+          "operation": "asyncAPI.yaml#consumeWait" 
+       }
+   ]
+}
+----
+
+Once the function is defined, you can invoke it within an `operation` state 
+
+.Example of an AsynyAPI subscribing function invocation
+[source,json]
+----
+   {
+      "name": "waitForEvent",
+      "type": "operation",
+      "actions": [
+          { 
+           "functionRef": "consumeEvent"
+          }
+       ]      
+    }
+----
+
+When your flow reach `waitForEvent` state, it will stop till it receives a CloudEvent published over Kafka topic `wait`. 
+
+[NOTE]
+====
+The URI part of `operation` property supports `http`, `https` and `file` schemas. If no schema is present, it assumes the specification file is on project classpath
+====
+
+== Publishing an event
+
+Lets assume there is an AsyncAPI specification file called `asyncAPI.yaml`, located in the classpath of your project. 
+
+This files contains a server definition which protocol is HTTP 
+
+.Example of an AsyncAPI http server definition
+[source,yaml]
+----
+servers:
+  development:
+    url: localhost:8080
+    description: Development server
+    protocol: http
+    protocolVersion: '1.0.0'
+----
+
+and a publishing channel named `resume`
+
+.Example of an AsynyAPI publishing channel definition
+[source,yaml]
+----
+channels:
+  resume:
+    description: A message channel
+    publish:
+      operationId: sendResume
+      summary: Get messages
+      message:
+        $ref: '#/components/messages/message'
+----
+
+You can define a function in your workflow, of type asyncapi, that point to `sendResume` operationId.
+
+.Example of an AsynyAPI publishing function definition
+[source,json]
+----
+{
+   "functions":[
+      {
+          "name": "sendResume",
+          "type": "asyncapi",
+          "operation": "asyncAPI.yaml#sendResume" 
+       }
+   ]
+}
+----
+
+Once the function is defined, you can invoke it within an `operation` state 
+
+.Example of an AsynyAPI publishing function invocation
+[source,json]
+----
+   {
+      "name": "sendEvent",
+      "type": "operation",
+      "actions": [
+          { 
+           "functionRef": "sendResume",
+           "arguments" : {
+              "name" : "Javierito"
+            }
+          }
+       ]      
+    }
+----
+
+When your flow reach `sendEvent` state, it will send an event to endpoint `http://localhost:8080/resume`, with payload `{"name":"Javierito"}`
```

Comment:
```suggestion
When your flow reach the `sendEvent` state, it will send an event to the endpoint `http://localhost:8080/resume`, with the payload `{"name":"Javierito"}`
```